Rubberduck.Main/VbeProvider.cs

```diff
@@ -16,19 +17,23 @@ namespace Rubberduck
     /// </remarks>
     internal static class VbeProvider
     {
-        internal static void Initialize(IVBE vbe)
+        internal static void Initialize(IVBE vbe, IVbeNativeApi vbeNativeApi, IBeepInterceptor beepInterceptor)
         {
             Vbe = vbe;
-            VbeRuntime = new VbeNativeApiAccessor();
+            VbeNativeApi = vbeNativeApi;
+            BeepInterceptor = beepInterceptor;
         }
 
         internal static void Terminate()
         {
             Vbe = null;
-            VbeRuntime = null;
+            VbeNativeApi = null;
+            BeepInterceptor?.Dispose();
+            BeepInterceptor = null;
```

Comment:
I hope the beep interceptor is configured in transient scope.

Reply:
It's configured as a Singleton, but that's just fine, considering the instance injected through CW is obained from here (and not the other way round)